test/test_esapi.py

```diff
@@ -1435,6 +1437,67 @@ def test_hmac(self):
         with self.assertRaises(TypeError):
             self.ectx.hmac(primaryHandle, inData, TPM2_ALG.SHA256, session3=45.6)
 
+    @pytest.mark.skipif(
+        not _lib_version_atleast("tss2-esapi", "3.2.0-167-gc17e3989"),
+        reason="Attached Components not supported prior to tss v4",
+    )
+    def test_mac(self):
+
+        attrs = (
+            TPMA_OBJECT.SIGN_ENCRYPT
+            | TPMA_OBJECT.USERWITHAUTH
+            | TPMA_OBJECT.SENSITIVEDATAORIGIN
+        )
+        templ = TPMT_PUBLIC.parse(alg="hmac", objectAttributes=attrs)
+        inPublic = TPM2B_PUBLIC(templ)
+
+        inSensitive = TPM2B_SENSITIVE_CREATE(TPMS_SENSITIVE_CREATE())
+
+        primaryHandle = self.ectx.create_primary(inSensitive, inPublic)[0]
+
+        # Test for tss v < 4.0.0
+        if not _lib_version_atleast("tss2-esapi", "4.0.0"):
+            with self.assertRaises(NotImplementedError):
+                self.ectx.mac(primaryHandle, b"1234", TPM2_ALG.SHA256)
+            return
```

Comment:
Is this test block ever running, the NotImplementedError is never thrown by ESAPI python code.